src/Runtime/CMakeLists.txt

```diff
@@ -114,3 +114,24 @@ llvm_update_compile_flags(PyRuntime)
 install(TARGETS PyRuntime
   DESTINATION lib
   )
+
+pybind11_add_module(PyCompileAndRuntime PyCompileExecutionSession.cpp)
+add_dependencies(PyCompileAndRuntime onnx_proto)
+target_compile_definitions(PyCompileAndRuntime
+  PRIVATE
+  $<TARGET_PROPERTY:onnx,COMPILE_DEFINITIONS>
+  )
+target_include_directories(PyCompileAndRuntime
+  PRIVATE
+  ${ONNX_MLIR_SRC_ROOT}/include
+  )
+target_link_libraries(PyCompileAndRuntime
+  PRIVATE
+  OnnxMlirCompiler
+  ExecutionSession
```

Comment:
@AlexandreEichenberger Hello, Alexandre. I seem to be stuck here. The whole project can compile with no issues. However, it will throw an ImportError with the message `undefined symbol` when trying to load it in Python. The whole error message is:
`ImportError: /workdir/onnx-mlir/docs/mnist_example/PyCompileAndRuntime.cpython-38-x86_64-linux-gnu.so: undefined symbol: _ZN9onnx_mlir18PyExecutionSession15pySetEntryPointENSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE`

The `undefined symbol` when importing sounds like the built module isn't being linked to the library, which is related to the `target_link_libraries` in the `CMakeLists.txt`. I have tried to remove the `OnnxMlirCompiler` in this `CMakeLists.txt` and it gives the error:
`ImportError: /workdir/onnx-mlir/docs/mnist_example/PyCompileAndRuntime.cpython-38-x86_64-linux-gnu.so: undefined symbol: _ZN9onnx_mlir18PyExecutionSession16pyInputSignatureB5cxx11Ev` which consistent with my guess.

Now the problem is that the error currently I meet is related to the function `pySetEntryPoint()` which is a member in `PyExecutionSession` class. From my understanding, the type `MODULE_LIBRARY` cannot be linked into another target. One may link only to `INTERFACE`, `OBJECT`, `STATIC` or `SHARED` libraries, or to executables with the `ENABLE_EXPORTS` property set. I am not sure whether there is any way to link the `PyRuntime` (type `MODULE_LIBRARY`) which contains the `PyExecutionSession` class with the new class `PyCompileAndRuntime` or maybe I need to find another way to solve this problem. Do you have any ideas or suggestions?